### ISSUE TITLE
Widget type switches when switching between backends

### ISSUE THREAD

**Issue by philippjfr** [ioam/holoviews#259]
When switching backends using the cell magic it switches to the scrubber widget for no apparent reason.

```python
import numpy as np
import holoviews as hv

%load_ext holoviews.ipython
%output backend='bokeh'

%%output backend='matplotlib' holomap='widgets'
hv.HoloMap({i: hv.Curve(np.random.rand(10,2)) for i in range(10)})
```

The third cell always uses scrubber widgets the first time you execute it here whether you specify ``holomap='widgets'`` or not.

**Comment by jlstevens** [comment]
Ok, I'll look at this as soon as I have a chance.

**Comment by philippjfr** [comment]
This was fixed by #308. Closing.
